source/thisTarget.ts

```diff
@@ -5,16 +5,20 @@ import {
 } from "webext-detect-page";
 import { messenger } from "./sender.js";
 import { registerMethods } from "./receiver.js";
-import { AnyTarget, MessengerMeta, Sender } from "./types.js";
-import { debug } from "./shared.js";
+import { AnyTarget, Message, MessengerMeta, Sender } from "./types.js";
+import { debug, MessengerError } from "./shared.js";
 import { Entries } from "type-fest";
 
 // Soft warning: Race conditions are possible.
 // This CANNOT be awaited because waiting for it means "I will handle the message."
 // If a message is received before this is ready, it will just have to be ignored.
-let thisTarget: AnyTarget | undefined = isBackground()
+const thisTarget: AnyTarget = isBackground()
   ? { page: "background" }
-  : undefined;
+  : {
+      get page(): string {
+        return location.pathname + location.search;
```

Comment:
If the URL were to be changed dynamically, this would now updated too.

Reply:
Doesn't setting this to a defined value introduce race conditions on the target comparison logic in compareTargets?

I guess that's the intent? You want to replace the "A message was received before this context was ready" warnings will now be replaced with "ignored due to target mismatch" errors?

Reply:
> introduce race conditions

As the comment mentioned, this already had race conditions. This doesn't affect that one nor should it introduce more, from what I see.

> warnings will now be replaced with "ignored due to target mismatch" debugging statements?

Correct, I noticed that they effectively mean the same thing. The "mismatch" debug message now includes more information regarding the status of `thisTarget` query, so they're effectively equivalent.

This change enables:

- a message with target `{page: "/options.html"}` can be handled immediately even before the tab information is ready, because the target doesn't request a specific tabId